File: golang/read_file.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	filePath := "./my-text-file.txt"

	file, _ := os.Open(filePath)
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		if line != "" {
			fmt.Println("line:", line)
		}
	}
}

File: cplusplus/read_file.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream file("./my-text-file.txt");

    for (std::string line; std::getline(file, line); )
        std::cout << "line: " << line << "\n";

    return 0;
}
